src/infrastructure/acknowledgement.rs

```diff
@@ -0,0 +1,241 @@
+use crate::packet::OrderingGuarantee;
+use crate::sequence_buffer::SequenceBuffer;
+use crate::packet::SequenceNumber;
+use std::collections::HashMap;
+
+const REDUNDANT_PACKET_ACKS_SIZE: u16 = 32;
+
+/// Responsible for handling the acknowledgement of packets.
```

Comment:
```suggestion
/// Responsible for handling the acknowledgment of packets.
```

Reply:
Heh surprised to see this here since it's common to include the e outside of the US. I can change it throughout if you want.

Reply:
hrrmm, I didn't even know that there were multiple ways to write this, my grammar checkers just came up with this. Just do what you think that's correct English :)

Reply:
I'll change it across the board with the new spelling. 👍